eox_core/api/support/v1/views.py

```diff
@@ -127,3 +138,90 @@ def patch(self, request, *args, **kwargs):
             user, custom_fields=admin_fields, context={"request": request}
         )
         return Response(serialized_user.data)
+
+
+class OauthApplicationAPIView(UserQueryMixin, APIView):
+    """
+    Handles requests related to the
+    django_oauth_toolkit Application object.
+    """
+    authentication_classes = (BearerAuthentication, SessionAuthentication)
+    permission_classes = (EoxCoreSupportAPIPermission,)
+    renderer_classes = (JSONRenderer, BrowsableAPIRenderer)
+
+    @audit_drf_api(action='Generate Oauth Application.', method_name='eox_core_api_method')
+    def post(self, request, *args, **kwargs):  # pylint: disable=too-many-locals
+        """
+        Created a new Oauth Application from django_oauth_toolkit.
+
+        In order to generate a valid application, this endpoint
+        does multiple things:
+        1 - Get or Create an edxapp user which will be the owner
+        of the Application.
+        2 - Grant permissions to the user that were sent in the
+        "permissions" list.
+        3 - Create Application.
+
+        Note: make sure to send the codename value of the permission
+        in the list.
+
+        For example:
+
+        **Requests**:
+            POST <domain>/eox-core/support-api/v1/oauth-application/
+
+        **Request body**:
+        {
+            "user": {
+                "fullname": "John Doe",
+                "email": "johndoe@example.com",
+                "username": "johndoe",
+                "permissions": ["can_call_eox_core", "can_call_eox_tenant"]
+            },
+            "redirect_uris": "http://testing-site.io/ http://testing-site.io",
+            "client_type":"confidential",
+            "authorization_grant_type":"client-credentials",
+            "name": "test-application",
+            "skip_authorization": true
+        }
+
+        **Response values**:
+            - 200: Success, the Oauth Applicaton has been created.
+            - 400: Bad request, invalid request body format.
+            - 500: The server has failed to get or create the
+            owner user for the application.
+        """
+        message = "Could not get or create edxapp User"
+
+        serializer = OauthApplicationSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        data = serializer.validated_data
+        user_creation_data = data.pop('user', {})
+        user_permissions = user_creation_data.pop('permissions', [])
+
+        user_creation_data.update({
+            'skip_extra_registration_fields': True,
+            'activate_user': True,
+            'skip_password': True,
+            'site': get_current_site(request),
+        })
+
+        # Get or create user
+        try:
+            user = get_edxapp_user(**user_creation_data)
+        except (NotFound, User.DoesNotExist):
+            user, message = create_edxapp_user(**user_creation_data)
```

Comment:
Is there not a get_or_create?

Reply:
@Alec4r nop, there is no get_or_create function to register a user in edxapp in the [backend module](https://github.com/eduNEXT/eox-core/blob/master/eox_core/edxapp_wrapper/backends/users_m_v1.py)